cmd/verify/verify.go

```diff
@@ -102,7 +110,13 @@ func main() {
 			if err != nil {
 				log.Error(err)
 			}
-			f := storage.FileReader(config.Archive.Type, filepath.Join(filepath.Clean(config.Archive.Location), message.ArchivePath))
+
+			var f io.Reader
+			if config.ArchiveType == "posix" {
+				f = backend.ReadFile(filepath.Join(filepath.Clean(config.PosixArchive.Location), message.ArchivePath))
+			} else {
+				f = backend.ReadFile(message.ArchivePath)
+			}
```

Comment:
See comment above.

Reply:
The backend here should know everything about how to construct the filepath. `verify.go` should only setup the correct backend at the start and then never care about it again, if the POSIX backend need to do some filepath joining or something, that should be done inside the `ReadFile` function so these lines can be only:

```go
f := backend.ReadFile(message.ArchivePath)
```

Reply:
Done